### ISSUE TITLE
In multi-select, selections do not appear in the order in which they were selected (solution)

### ISSUE THREAD

**Issue by gwcms** [select2/select2#5441]
//to prevent auto ordering after select
$('#yourElementId').on("select2:select", function (evt) {
	var element = evt.params.data.element;
	var $element = $(element);
	$element.detach();
	$(this).append($element);
	$(this).trigger("change");
});	
			
			
//allow ordering			
$.fn.select2.amd.require([
	'select2/utils',
	'select2/dropdown',
	'select2/dropdown/attachBody'
], function (Utils, Dropdown, AttachBody) {
	var select = $('#yourElementId');
	var container = select.select2().data('select2').$container;
	
	container.find('ul').sortable({
	containment: 'parent',
	update: function(event, ui) { 				
		$(ui.item[0]).parent().find('.select2-selection__choice').each(function(){ 
			var elm = Utils.__cache[$(this).data('select2Id')];
			var element = elm.data.element;
			var $element = $(element);
			$element.detach();
			select.append($element);
			select.trigger("change");
		})
		}
	});	
})

**Comment by yackers** [comment]
Thanks for solution, but I have another question in that case. So we can sort when selecting, but how to be with order of initial selected (default) values ?
Because this is not working properly:

`$('#mySelect2').val(['2', '1']);
$('#mySelect2').trigger('change');`

Result always will be  = 1, 2; 

Any solution ?
